core/store/src/node_storage/mod.rs

```diff
@@ -198,8 +198,8 @@ impl NodeStorage {
         self.cold_storage.is_some()
     }
 
-    /// Reads database metadata and returns whether the storage is archival.
-    pub fn is_archive(&self) -> io::Result<bool> {
+    /// Reads database metadata and returns whether it is split storage or legacy archival node.
+    pub fn is_cold_archive(&self) -> io::Result<bool> {
```

Comment:
Again the name is confusing. Perhaps we need a new term for a split or legacy node? Something like `is_local_archive` and complement it with `is_cloud_archive`? 

```
local := legacy || split
cloud := cloud writer || cloud reader
```